cpp/include/cuspatial/vec_2d.hpp

```diff
@@ -98,6 +107,68 @@ vec_2d<T> CUSPATIAL_HOST_DEVICE operator*(T const& r, vec_2d<T> vec)
   return vec * r;
 }
 
+/**
+ * @brief Translate a 2D point
+ */
+template <typename T>
+vec_2d<T>& CUSPATIAL_HOST_DEVICE operator+=(vec_2d<T>& a, vec_2d<T> const& b)
+{
+  a.x += b.x;
+  a.y += b.y;
+  return a;
+}
+
+/**
+ * @brief Translate a 2D point
+ */
+template <typename T>
+vec_2d<T>& CUSPATIAL_HOST_DEVICE operator-=(vec_2d<T>& a, vec_2d<T> const& b)
+{
+  return a += -b;
+}
+
+/**
+ * @brief Less than operator for two 2D points.
+ *
+ * Orders two points first by x, then by y.
+ */
+template <typename T>
+bool CUSPATIAL_HOST_DEVICE operator<(vec_2d<T> const& lhs, vec_2d<T> const& rhs)
+{
+  if (lhs.x < rhs.x)
+    return true;
+  else if (lhs.x == rhs.x)
+    return lhs.y < rhs.y;
+  return false;
+}
+
+/**
+ * @brief Greater than operator for two 2D points.
+ */
+template <typename T>
+bool CUSPATIAL_HOST_DEVICE operator>(vec_2d<T> const& lhs, vec_2d<T> const& rhs)
```

Comment:
You could improve your compile time by making these operators hidden friends: https://www.justsoftwaresolutions.co.uk/cplusplus/hidden-friends.html

Reply:
Moved. I think a 2nd benefit of this is that has @harrism [pointed out](https://github.com/rapidsai/cuspatial/pull/778#pullrequestreview-1172981032), that the definition of these comparators is not canonical, and we want to avoid users who choose to use type convertible to `vec_2d` to misuse these operator. Making them hidden friend requires user to explicitly cast them before they can access these operators.